### PR TITLE
add token verification

### PR DESCRIPTION
**Pull request by ctgnauh**
Hi @pchaigno 

First of all, please forgive my poor english. I try to make it clear.

A few days ago, i also wrote a hubot script almost like this. But yours better!

Sometimes i need use ifttt to connect hubot and other web services, i think a security measure is necessary.

So i added a simple token verification for hubot-notify.

### CODE REVIEW

File: src/notify.coffee [hunk 1]
```diff
@@ -15,6 +15,9 @@
 
 module.exports = (robot) ->
   robot.router.post '/notify/:target', (req, res) ->
+    if req.body.token isnt process.env.NOTIFY_TOKEN or ''
```
Comment (pchaigno) [inline src/notify.coffee]: I think we should first check that `process.env.NOTIFY_TOKEN` exists for backward compatibility:

``` coffee
if process.env.NOTIFY_TOKEN? and req.body.token isnt process.env.NOTIFY_TOKEN
```

So if `NOTIFY_TOKEN` is undefined, it accepts any request.

File: src/notify.coffee [hunk 2]
```diff
@@ -15,6 +15,9 @@
...
+      res.status(400).send 'error: token verification failed\n'
```
Comment (pchaigno) [inline src/notify.coffee]: I think a 401 would be more appropriate here.

**Comment by pchaigno** [comment]
Thanks for the pull request @ctgnauh!
I'm not exposing my Hubot on the Internet but that was still one of the things I wanted to add :-)

`NOTIFY_PASSWORD` might be more appropriate than `NOTIFY_TOKEN` here since it doesn't have an expiration time. What to you think?

Also, would you be willing to add a test for your changes? I can do it afterwards if you prefer.

File: src/notify.coffee (hunk 2 above)
Comment (ctgnauh) [inline src/notify.coffee]: OK.

I didn't do anything right :-(

**Comment by ctgnauh** [comment]
No, it's not a real token. I think NOTIFY_KEY or NOTIFY_SECRET may be better.

Yes, i will add a test for my code. I'm glad to do this.

File: src/notify.coffee (hunk 2 above)
Comment (pchaigno) [inline src/notify.coffee]: That's not true. Those are minor tweaks :-)

**Comment by pchaigno** [comment]
> No, it's not a real token. I think NOTIFY_KEY or NOTIFY_SECRET may be better.

Okay for `NOTIFY_SECRET`. That'd be consistent with other APIs (GitHub's for instance).

> Yes, i will add a test for my code. I'm glad to do this.

Awesome! If you need help, feel free to ask :smiley:

**Comment by ctgnauh** [comment]
Thank you, I learned a lot from this conversation.

**Comment by pchaigno** [comment]
I'm glad you did :smile: 
The test are great! Merging!

**Pull request merged**